### DISCUSSION TITLE
LUKS and Argon2:  Is this Debian's responsibility or mine?

### DISCUSSION THREAD

**Discussion by birds-swim** [SpiralLinux/SpiralLinux-project#260] [Q&A] [answered]
https://mjg59.dreamwidth.org/66429.html

I was reading this article today, and I wanted to know if LUKS2 with Argon2 will be available for SpiralLinux 12.  Or, if that's the user's responsibility to upgrade.

**Comment by geckolinux** [comment, marked as answer]
Hi there, I don't know anything about the internals of LUKS and I don't know what Argon is. 

As a simple rule, by design SpiralLinux (me) has no way to update or reconfigure anything on your system after installation. All updates to systems installed from a SpiralLinux ISO come from Debian. Debian in theory could include a postinst script in their DEB package to make `apt` or Synaptic show a prompt asking the user how to proceed during the upgrade, or they could simply silently upgrade and change the config.

Where SpiralLinux comes into play is putting together the ISO and during the system installation. The installation part depends on the Calamares defaults and/or any Calamares config overrides that I apply. I haven't customized the LUKS defaults at all for Calamares, so it would be using whatever behavior Calamares ships by default.

**Reply by birds-swim** [reply]
And therefore whatever defaults Debian is using, correct?

Thank you for your reply.  I am looking forward to the Bookworm release!
